### DISCUSSION TITLE
Lock bucket for specific users

### DISCUSSION THREAD

**Discussion by happyDuke96** [vladimir-bukhtoyarov/bucket4j#383] [Q&A] [answered]
Hello, I'm completely new to bucket4j, I need to catch every request to the specific api, if the api is called more than 100 times in 1 hour, the API will be blocked for 2 hours for the specific user, after unlocking there will be some other limit not like the previous one, and so on, and I want to dynamically change this setting. Is there a way to implement this with Bucket4j. I looked at the documentation, it basically says how to save these buckets.Any help is appreciated

**Comment by happyDuke96** [comment]
Hello,@vladimir-bukhtoyarov. 
If there is a way to do this, can you show a minimal example?

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
> Is there a way to implement this with Bucket4j.

Out-of-the-box no. Bucket4j just implements token-bucket algorithm. It can provide only simple answer about available tokens. Implementation of thotling and penalty strategis is out of scope, you should implement it by yourself. Anyway dynamic reconfiguration [is supported](https://bucket4j.com/8.3.0/toc.html#configuration-replacement). Blocking of bucket for specified period of time is not supported, but  you can aproximatelly emulate this behavior by [force consumption](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/Bucket.java#L95).

> If there is a way to do this, can you show a minimal example?

No. I have no time to do so, try to start from there https://bucket4j.com/#third-party-articles and ask more concrete questions

**Reply by happyDuke96** [reply]
Thank you for your time
